pkg/test/devstack/default_publisher_test.go

```diff
@@ -74,7 +72,20 @@ func (s *DefaultPublisherSuite) TestDefaultPublisher() {
 				},
 			},
 		},
-		Stack:          &stack,
+		// TODO(unsure): inorder for the default publisher to apply, the task needs ResultPaths set with no publisher.
+		// We can't results paths directly on the task because scenarios use this field to override tasks result paths.
+		// So here we are setting the ResultsPath on the above task.
+		// By omitting the publisher from the task we allow allow the orchestrator to populate it with job defaults,
+		// which sets the default publisher.
+		// However, in https://github.com/bacalhau-project/bacalhau/pull/2802/files#diff-7c0b9e1e39b2a2f5a00159ab3b4bed52d426dfa345498495ee8ee69dee6cc56aR130-R131
+		// validation was added the marks tasks with result paths and no publisher as invalid, so this test can't run
+		// and thus we skip it.
```

Comment:
I don't get this. You are saying scenarios override tasks result paths, which means they are setting a result path on our behalf, and then the default publisher should kick in, right? 

If the change is trivial to still make this work, then lets please do it. We've had a lot of cases where we skip tests and never fix them

Reply:
Here: https://www.notion.so/Rethinking-Configuration-435fbe87419148b4bbc5119d413786eb?d=1c73cb661b18405b8b67b3856f02e1d1&pvs=4#fdc6de7c00364de1bb81b325686e6e66 it was stated:

> Based on our conversation, the intention of the default publisher is to ONLY be used if the a client specifies a ResultsPath but doesn’t provide a publisher. If a client submits a job without a ResultPath and no publisher then the DefaultPublisher isn’t used

In this PR, the transformer that applies the default publisher is inline with these requirements, only setting a default publisher if the task specified a results path without a publisher.:
https://github.com/bacalhau-project/bacalhau/blob/frrist/config/new-type/pkg/orchestrator/transformer/job.go#L73

So this is the reason I am including a ResultsPath in the task submitted here - aiming to simulate the submission of a task without a publisher defined and with ResultsPath set.

The previous implementation did not take result paths into account when setting the publisher: https://github.com/bacalhau-project/bacalhau/blob/main/pkg/orchestrator/transformer/publisher.go#L13. I have removed this transformer as it breaks our requirements. Also note that it breaks this test.

In opposition to the requirements, current task validation considers tasks with result paths set and publisher unset to be invalid.
https://github.com/bacalhau-project/bacalhau/blob/d640a413482a9f6801a944a3dc77e7662776dd86/pkg/models/task.go#L138-L140
The validation is enforced in several different places, both on the client side and the api (server) side. 
This means the situation in which a job should receive a default publisher will never arise given the requirements in the Rethinking Configration doc. Since If a job is submitted with ResultPaths set but no publisher set it will be considered invalid before the logic in the transformer can be applied.

Reply:
Imho the simplest thing to do here would be abandon the notion of a default publisher. If a client submits a job that ommits the publisher and sets the ResultsPath the network should reject it with a helpful error.

Reply:
The simplest thing to do is to just maintain the existing behaviour. We can think about changing it in a separate PR and task, but not as part of this large and time critical PR. We should not remove the concept of default publisher, and should not apply the default publisher conditionally like you are doing here